src/main/java/com/glodblock/github/common/tile/TileSuperStockReplenisher.java

```diff
@@ -0,0 +1,462 @@
+        return IntStream.rangeClosed(0, 62).toArray();
+    }
+
+    @Nonnull
+    @Override
+    public IInventory getInternalInventory() {
+        return invItems;
+    }
+
+    public AppEngInternalAEInventory getConfigFluid() {
+        return configFluids;
+    }
+
+    public AppEngInternalAEInventory getConfigItems() {
+        return configItems;
+    }
+
+    public AppEngInternalInventory getCell() {
+        return cell;
+    }
+
+    public long getFreeBytes() {
+        return totalBytes - (((storedItemCount + getUnusedItemCount()) / 8)
+                + ((storedFluidCount + getUnusedFluidCount()) / 2048));
+    }
+
+    public long getRemainingFluidCount() {
+        final double remaining = (double) getFreeBytes() * 2048 + this.getUnusedFluidCount();
+        if (remaining > Long.MAX_VALUE) return Long.MAX_VALUE;
+        return remaining > 0 ? (long) remaining : 0;
+    }
+
+    public int getUnusedFluidCount() {
+        final int div = (int) (storedFluidCount % 2048);
+        if (div == 0) {
+            return 0;
+        }
+        return 2048 - div;
+    }
+
+    public long getRemainingItemCount() {
+        final long remaining = getFreeBytes() * 8 + this.getUnusedItemCount();
+
+        return remaining > 0 ? remaining : 0;
+    }
+
+    public int getUnusedItemCount() {
+        final long div = storedItemCount % 8;
+
+        if (div == 0) {
+            return 0;
+        }
+
+        return (int) (8 - div);
+    }
+
+    @Override
+    public void onChangeInventory(IInventory inv, int slot, InvOperation mc, ItemStack removed, ItemStack added) {
+        switch (mc) {
+            case setInventorySlotContents -> {
+                if (added != null) {
+                    if (inv == cell) {
+                        if (added.getItem() instanceof ItemBasicStorageCell ibsc) {
+                            totalBytes = ibsc.getBytesLong(added);
+                        } else if (added.getItem() instanceof FCBaseItemCell fcbic) {
+                            totalBytes = fcbic.getBytes(added);
+                        }
+                        getProxy().setIdlePowerUsage(Math.sqrt(Math.pow(totalBytes, 0.576D)));
+                    } else if (inv == invItems) {
+                        storedItemCount += added.stackSize;
+                    }
+                }
+
+                if (removed != null) {
+                    if (inv == cell) {
+                        totalBytes = 0;
+                        getProxy().setIdlePowerUsage(4d);
+                    } else if (inv == invItems) {
+                        storedItemCount -= removed.stackSize;
+                    }
+                }
+            }
+            case decreaseStackSize -> {
+                if (inv == cell) {
+                    totalBytes = 0;
+                    getProxy().setIdlePowerUsage(4d);
+                }
+                if (inv == invItems && invItems.getStackInSlot(slot) != null) storedItemCount -= removed.stackSize;
```

Comment:
If extracted by a hopper or other device and the stack size becomes 0, the stack in the slot will become null.
```suggestion
                if (inv == invItems) storedItemCount -= removed.stackSize;
```

Reply:
done